backend/ddang/src/main/java/com/ddang/ddang/chat/domain/WebSocketChatSessions.java

```diff
@@ -0,0 +1,48 @@
+package com.ddang.ddang.chat.domain;
+
+import lombok.Getter;
+import org.springframework.stereotype.Component;
+import org.springframework.web.socket.WebSocketSession;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+@Component
+@Getter
+public class WebSocketChatSessions {
+
+    private static final String ATTRIBUTE_KEY = "chatRoomId";
+    private final Map<Long, WebSocketSessions> chatRoomSessions;
+
+    public WebSocketChatSessions() {
+        this.chatRoomSessions = new ConcurrentHashMap<>();
+    }
+
+    public void add(final WebSocketSession session, final Long chatRoomId) {
+        chatRoomSessions.putIfAbsent(chatRoomId, new WebSocketSessions());
+        final WebSocketSessions webSocketSessions = chatRoomSessions.get(chatRoomId);
+        if (!webSocketSessions.containsValue(session)) {
+            webSocketSessions.add(session);
+            session.getAttributes().put(ATTRIBUTE_KEY, chatRoomId);
+        }
```

Comment:
```suggestion
       webSocketSessions.putIfAbsent(session);
```

WebSocketSessions가 직접 존재하는지 확인하는 것과 세션을 추가해주는 메서드를 생성하고
분기문을 생성한 webSocketSessions의 메서드로 대체하는건 어떤가요?

Reply:
`Map`에서는 가능했지만, `Set`에서도 가능한 건가요?

Reply:
putIfAbsent() 메서드를 webSocketSessions 일급 컬렉션에 만들고 if문을 webSocketSessions 내부로 이동시키는 것이 어떤지에 대한 의견이었습니다
그런데 이건 취향 차이일 수도 있을 것 같아 제이미의 선택에 맡길게요!

Reply:
set에도 해당 메서드가 있다는 의미로 잘못 이해했네요..!
메리의 의견 이해했습니다. 좋은 의견인 것 같아요!
해당 방식으로 수정했습니다.